### PR TITLE
Add new rule socket_systemd-journal-remote_disabled

### PR DESCRIPTION
**Pull request by dodys**
#### Description:

- Rule to disable `systemd-journal-remote.socket`

#### Rationale:

- This rule is needed on CIS for both Ubuntu and RHEL

### CODE REVIEW

File: linux_os/guide/system/logging/journald/service_systemd-journal-remote_disabled/rule.yml
Comment (vojtapolasek) [inline linux_os/guide/system/logging/journald/service_systemd-journal-remote_disabled/rule.yml:14]: I see that you are aiming at the "socket" unit.
I suggest using the following macro: https://github.com/ComplianceAsCode/content/blob/e227941887ea63aa060c81368d24ffe712e1367c/shared/macros/01-general.jinja#L289

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello, thank you for the new rule. See my comments. But here is the main question:
Are you aiming at disabling ONLY the socket as stated in CIS benchmark?
If yes, the you should know that themplate you are using disables the service unit as well as the socket unit.

File: linux_os/guide/system/logging/journald/service_systemd-journal-remote_disabled/rule.yml
Comment (dodys) [reply linux_os/guide/system/logging/journald/service_systemd-journal-remote_disabled/rule.yml:12]: Yeah, that text comes from CIS itself, and I agree that calling both as services is a bit odd, will change it to systemd unit files as suggested

File: linux_os/guide/system/logging/journald/service_systemd-journal-remote_disabled/rule.yml
Comment (dodys) [reply linux_os/guide/system/logging/journald/service_systemd-journal-remote_disabled/rule.yml:14]: I was actually unsure about this, they only mention the socket, but could the service somehow enable the socket. I still haven't checked their unit files tbh

**Comment by vojtapolasek** [comment]
@dodys  are you planing to add test scenarios as well? How about Ansible remediation?

**Comment by dodys** [comment]
> @dodys are you planing to add test scenarios as well? How about Ansible remediation?

Just pushed tests for it.
Regarding ansible I won't be pushing anything for it, as we don't support it and have no way to test it properly

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/bash/shared.sh
Comment (vojtapolasek) [inline linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/bash/shared.sh:3]: This really only disables the service. But we want it masked. I think you will not be able to use existing macro.
you can either create a new macro for masking of something or you can just not use macro.
I think the best inspiration is here:
https://github.com/ComplianceAsCode/content/blob/7a0f2e0db6eda8d3a153c71779d8e636028d55d8/shared/templates/service_disabled/bash.template#L9

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml
Comment (vojtapolasek) [inline linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml:2]: I think this rule deserves a prodtype. If nothing, then the service / package seems not to exist on RHEL 7. Please add desired ubuntu prodtypes, other products can be added later.

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml
Comment (vojtapolasek) [inline linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml:24]:
```suggestion
ocil_clause: 'the systemd-journal-remote socket is not masked'
```

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml
Comment (vojtapolasek) [inline linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml:27]: I am afraid this macro is not suitable for this case. It is suitable for services. The problem is that it mixes terminology - you put there the parameter with .socket extension, but the text provided by macro talks about services. Sockets are not services - they are Systemd units, but AFAIK socket is not a service nor subset of service. Socket is used to activate a service as a reaction to connection.
More info here:

https://unix.stackexchange.com/questions/159462/what-is-systemds-target-service-and-socket


You can inspect the final result of macro expansion by building your product and then looking into
```
build/<product>/rules/socket_systemd-journal-remote_disabled.yml
```
I think you will have to create a new macro or input the ocil manually.

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml
Comment (vojtapolasek) [inline linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml:30]: Here goes the same as for the ocil above. I think you need a new macro or input the fixtext manually so that the terminology is not mixed up.

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello,
thank you for updates. It wil need few more changes, see comments.

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml
Comment (dodys) [reply linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml]: created a new macro for it

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml
Comment (dodys) [reply linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml]: I didn't create one but instead used the available:
socket_disabled_check_with_systemd

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/bash/shared.sh
Comment (vojtapolasek) [inline linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/bash/shared.sh:3]: I think this line should be removed. Installation of the package is definitely out of scope for this rule.

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello, thank you for improvements. Just two more things and I think we are ready to merge it.

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml
Comment (dodys) [reply linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/rule.yml:3]: oops, fixed!

File: linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/bash/shared.sh
Comment (dodys) [reply linux_os/guide/system/logging/journald/socket_systemd-journal-remote_disabled/bash/shared.sh]: removed it

**Review by vojtapolasek — APPROVED** [review]
Looks good now.
automatus tests are failing because during the PR development, there were added test scenarios (which passed), but later it was decided to restrict the rule to Ubuntu2204 prodtype. I think it does not make sense to remove those tests - it is possible that the rule will be eventually reused in RHEL since it is from CIS profile.

**Comment by vojtapolasek** [comment]
@dodys  are you able to approve this PR? I can't merge it because @ComplianceAsCode/ubuntu-maintainers  is required to approve the PR and you are the only member.@dodys  are you planing to add test scenarios as well? How about Ansible remediation?

**Comment by dodys** [comment]
> @dodys are you able to approve this PR? I can't merge it because @ComplianceAsCode/ubuntu-maintainers is required to approve the PR and you are the only member.@dodys are you planing to add test scenarios as well? How about Ansible remediation?

@vojtapolasek no I cannot approve PRs that I'm the author. @marcusburghardt can you help us here?
also @vojtapolasek regarding the test and ansible I've mentioned before:
"Just pushed tests for it.
Regarding ansible I won't be pushing anything for it, as we don't support it and have no way to test it properly"

**Review by marcusburghardt — APPROVED** [review]
Thanks for this rule. It will also be useful for other distros. I will be happy to include the Ansible remediation for it.

**Comment by marcusburghardt** [comment]
Automatus CS8, CS9 and Fedora are failing because the rule is restricted to `ubuntu2204` in its `prodtype`. Waived.
`ci/prow/e2e-aws-ocp4-moderate-node` failed due to test environment issue but can safely be waived too since the rule is specific for `ubuntu2204`.

**Comment by marcusburghardt** [comment]
Overriding CODEOWNERS since @dodys can't approve his own PR.

**Pull request merged**
